docs/testplan/smart-switch/high-availability/keysight_ha_testplan.md

```diff
@@ -0,0 +1,221 @@
+
+### Keysight Testbed
+Tests will run on the following testbeds.
+
+![HA_TestBed](images/ha_testbed.svg)
+
+Model expanded to show Keysight chassis, UHD, and SmartSwitch.  The UHD and SmartSwitchs can be used to conifgure various test case scenarios and network configurations.  For example, simulating DPUs within the same SmartSwitch or pairing distant DPUs to form HA sets.  The UHD has the ability to be configured in such a way to simulate individual ToRs.  The UHD shall additionally be used with Stateful traffic for VxLan encapsulation and decapsulation.
+
+![HA_SmartSwitch_Topology](images/smartswitch_ha_topology.svg)
+
+## Topology
+### Configuration for HA testing
+![HA_topology](images/ha_test_topology.svg)
+
+## Setup Configuration
+Two DPUs will use similar network configurations (VxLAN, ENI) forming an HA set.  One DPU will be designated as the Active the second Standby.
+
+## Test Methodology
+Following test methodology will be used for measuring HA switchover and performance.
+* Traffic generator will be used to configure ENI peering between DPU ports.
+* Data traffic will be sent from  server to server, server to T1 and T1 to server.
+* Depending on the test case, switchovers/failovers will be generated and measured.
+* Switchover between DPUs will be measured by noting down the precise time of the switchover/failover event.  Traffic generator will create those timestamps and provide us with the recovery statistics.
+
+
+## Test cases
+
+### Test Metrics
+Each test case will use the following metrics to measure performance of the HA failover/switchover for all test cases.
+
+The following metrics shall be collected Before and After switchover is initiated.
+
+| Performance Metric                  | Achieved |
+|:------------------------------------|:--------:|
+| Concurrent Connections              |         |
+| CPS                                 |         |
+| PPS                                 |         |
+| TCP Client Failures Resets Sent     |         |
+| TCP Client Failures Resets Recieved |         |
+| TCP Client Failures Resets Retries  |         |
+| TCP Server Failures Resets Sent     |         |
+| TCP Server Failures Resets Recieved |         |
+| TCP Server Failures Resets Retries  |         |
+
+As soon as failover/switchover is initiated a timer will be initiated to measure DPU pair switchover.
+
+| Failover/Switchover<br/>Type |                   Event                    | Convergence (s) |
+|:------------------------|:------------------------------------------:| :---: |
+|                         | Switchover time from <br/>Active-to-Standby DPU | 0  |
+
+### Test case1 Link Loss Local SmartSwitch 
+#### Test Objective 1
+In this scenario we will test the switchover between an Active DPU to it's paired Standby DPU.
+
+![HA_LinkLoss](images/ha_linkloss_test.svg)
```

Comment:
Thanks for updating the TOR and Switch part in the diagrams, Alberto! And I believe the DPU part is still kinda missing, do you mind to update the DPU part accordingly as well? After that, we can present to the group.

To be more specific, the topology should look like close to below, where:

- DPU and ToR are modeled together as a physical / logical smart switch. This will be the DUT in the topology.
- DPU will only paired with other DPUs on a different switch.

Then we can test failure cases on all moving parts in the system, with integration of sonic-mgmt integration:

- Which switch should act as root fanout
- How to connect DUT eventually to the root fanout
- How to connect DUT to management switch, so we can manage it, login and debug

I made a quick small graph to demonstrate what I mean in high level, but they might not be accurate in your setup:

![image](https://github.com/sonic-net/sonic-mgmt/assets/1533278/403ddf7a-b182-441b-85ee-2bac096f4b8a)

So 2 things in the diagram - SmartSwitch modeling and sonic-mgmt integration. Let's see if we can get at least the first item updated before presenting to the SmartSwitch group.